api/src/main/java/org/pentaho/metaverse/api/analyzer/kettle/annotations/AnnotationDrivenStepMetaAnalyzer.java

```diff
@@ -98,6 +121,70 @@ protected void customAnalyze( BaseStepMeta meta, IMetaverseNode rootNode ) throw
     rootNode.setProperties(
       annoFields.props()
         .collect( toMap( AnnotatedClassField::name, AnnotatedClassField::val ) ) );
+
+    // check for a subtrans and generate the graph
+    if ( meta instanceof StepWithMappingMeta ) {
+      linkParentFieldsToSubtrans( meta, rootNode, externalNodes );
+    }
+  }
+
+  private void linkParentFieldsToSubtrans( Object meta, IMetaverseNode rootNode,
```

Comment:
Looks like cyclomatic complexity jumped quite a bit with this method.  Can you see if any of the nested expressions below can be cleaned up?  And/or split the subtrans logic into a separate class that gets used by this AnnotationDrivenStepAnalyzer.
Also, would be good to add a test or two for subtrans.